ui/components/Paginator.js

```diff
@@ -0,0 +1,91 @@
+import React, { useState, useEffect } from 'react'
+
+export const Paginator = ({
+  pagination,
+  setPagination,
+  totalJobs,
+}) => {
+  const totalPages = Math.ceil(totalJobs / 10)
+  const currentPageNumber = totalJobs > 0 ? Math.floor(pagination.start / 10) + 1 : 0
+
+  const [pageNumberInputValue, setPageNumberInputValue] = useState(currentPageNumber)
+
+  useEffect(() => {
+    setPageNumberInputValue(currentPageNumber)
+  }, [currentPageNumber])
+
+  const handleClickPrevPage = (event) => {
+    if (pagination.start >= 10) {
+      setPagination({
+        start: pagination.start - 10,
+        end: pagination.end - 10,
+      })
+    }
+  }
+
+  const handleClickNextPage = (event) => {
+    if (currentPageNumber < totalPages) {
+      setPagination({
+        start: pagination.start + 10,
+        end: pagination.end + 10,
+      })
+    }
+  }
+
+  const handlePageNumberInputSubmit = (event) => {
+    if (event.key !== 'Enter') {
+      return
+    }
+
+    const inputValue = parseInt(pageNumberInputValue)
+    if (inputValue !== NaN) {
+      if (inputValue > 0 && inputValue <= totalPages) {
+        const startIndex = (inputValue - 1) * 10
+        const endIndex = startIndex + (Math.min(Math.max(totalJobs - 1, 0), 9))
+        setPagination({
+          start: (inputValue - 1) * 10,
+          end: endIndex,
+        })
+      } else {
+        alert(`Invalid page number. Please input a number between 1 and ${totalPages}`)
```

Comment:
nice error handling here!

Reply:
Felt the logic a bit convoluted there, but hope it can catch most edge cases!